openeo/rest/connection.py

```diff
@@ -55,6 +55,34 @@ def __init__(self, http_status_code: int = None,
         self.url = url
         super().__init__("[{s}] {c}: {m}".format(s=self.http_status_code, c=self.code, m=self.message))
 
+class Paginate:
+
+    def __init__(self, con, url: str = None, callback: Callable = None):
+        self.con = con
+        self.url = url
+        self.nextUrl = url
+        self.callback = callback
+        self.page = 0
+
+    def __iter__(self):
+        self.page = 0
+        self.nextUrl = self.url
+        return self
+
+    def __next__(self):
+        if self.nextUrl is None:
+            raise StopIteration
+        
+        response = self.con.get(self.nextUrl).json()
+        if isinstance(response['links'], list):
+            nextLink = next((link for link in response['links'] if link['rel'] == 'next'), None)
+            self.nextUrl = nextLink['href']
```

Comment:
will fail if nextLink is None (there was no next link)

Reply:
Fixed